### PR TITLE
APR-1528 - AdminService - Gateway - Mutiple issues

### PR DESCRIPTION
**Pull request by MakCom**

### CODE REVIEW

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/RoatpOrganisationCriminalComplianceChecksControllerTests.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ public void Criminal_compliance_check_returns_view(string gatewayPageId)
             viewResult.Should().NotBeNull();
             var viewModel = viewResult.Model as OrganisationCriminalCompliancePageViewModel;
             viewModel.Should().NotBeNull();
-            viewModel.PageTitle.Should().Be(CriminalCompliancePageConfiguration.Titles[gatewayPageId]);
+            //viewModel.PageTitle.Should().Be(CriminalCompliancePageConfiguration.Titles[gatewayPageId]);
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/RoatpOrganisationCriminalComplianceChecksControllerTests.cs:77]: If this assertion isn't needed can it be deleted rather than commented out?

File: src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpOrganisationCriminalComplianceChecksController.cs [hunk 2]
```diff
@@ -34,8 +34,8 @@ public async Task<IActionResult> GetCriminalCompliancePage(Guid applicationId, s
             var username = _contextAccessor.HttpContext.User.UserDisplayName();
             var viewModel = await _orchestrator.GetCriminalComplianceCheckViewModel(new GetCriminalComplianceCheckRequest(applicationId, gatewayPageId, username));
 
-            viewModel.PageTitle = CriminalCompliancePageConfiguration.Titles[gatewayPageId];
-
+            //viewModel.PageTitle = CriminalCompliancePageConfiguration.Titles[gatewayPageId];
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpOrganisationCriminalComplianceChecksController.cs:37]: If this assertion isn't needed can it be deleted rather than commented out?

File: src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayCriminalComplianceChecksOrchestrator.cs [hunk 3]
```diff
@@ -23,24 +23,28 @@ public class GatewayCriminalComplianceChecksOrchestrator : IGatewayCriminalCompl
...
+            //{ 
+            //    ApplicationId = request.ApplicationId, 
+            //    PageId = request.PageId,
+            //    ApplyLegalName = commonDetails.LegalName,
+            //    Ukprn = commonDetails.Ukprn,
+            //    Status = commonDetails.Status,
+            //    OptionPassText = commonDetails.OptionPassText,
+            //    OptionFailText = commonDetails.OptionFailText,
+            //    OptionInProgressText = commonDetails.OptionInProgressText
+            //};
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayCriminalComplianceChecksOrchestrator.cs:39]: If not needed can it be deleted rather than commented out?

File: src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayCriminalComplianceChecksOrchestrator.cs [hunk 4]
```diff
@@ -23,24 +23,28 @@ public class GatewayCriminalComplianceChecksOrchestrator : IGatewayCriminalCompl
...
             _logger.LogInformation($"Retrieving criminal compliance details for application {request.ApplicationId} page {request.PageId}");
 
+            var model = new OrganisationCriminalCompliancePageViewModel();
+            await model.PopulatePageCommonDetails(_applyApiClient, request.ApplicationId, request.PageId, request.UserName);
+            model.Caption = RoatpGatewayConstants.Captions.OrganisationsCriminalAndComplianceChecks;
+            model.Heading = CriminalCompliancePageConfiguration.Headings[request.PageId];
+            model.NoSelectionErrorMessage = CriminalCompliancePageConfiguration.NoSelectionErrorMessages[request.PageId];
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayCriminalComplianceChecksOrchestrator.cs:46]: Is it worth putting these 3 sets in PopulatePageCommonDetails and passing the values as parameters?  That way it'll prevent people (me! 😁) forgetting to set them.

File: src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/OrganisationCriminalCompliancePageViewModel.cs [hunk 5]
```diff
@@ -4,9 +4,9 @@ namespace SFA.DAS.AdminService.Web.ViewModels.Roatp.Gateway
 {
     public class OrganisationCriminalCompliancePageViewModel : RoatpGatewayPageViewModel
     {
-        public string ApplyLegalName { get; set; }
+        //public string ApplyLegalName { get; set; } // We had it in RoatpGatewayPageViewModel
 
-        public string PageTitle { get; set; }
+        //public string PageTitle { get; set; }
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/OrganisationCriminalCompliancePageViewModel.cs:9]: Delete rather than comment out.

File: src/SFA.DAS.AssessorService.ApplyTypes/Roatp/CriminalCompliancePageConfiguration.cs [hunk 6]
```diff
@@ -4,12 +4,46 @@ namespace SFA.DAS.AssessorService.ApplyTypes.Roatp
...
     {
-        public static Dictionary<string, string> Titles = new Dictionary<string, string>
+        //public static Dictionary<string, string> Titles = new Dictionary<string, string>
+        //{
+        //    { GatewayPageIds.CriminalComplianceOrganisationChecks.CompositionCreditors, "Composition with creditors check" },
+        //    { GatewayPageIds.CriminalComplianceOrganisationChecks.FailedToRepayFunds, "Failed to pay back funds in the last 3 years check" },
+        //    { GatewayPageIds.CriminalComplianceOrganisationChecks.ContractTermination, "Contract terminated early by a public body in the last 3 years check" },
+        //    { GatewayPageIds.CriminalComplianceOrganisationChecks.ContractWithdrawnEarly, "Withdrawn from a contract with a public body in the last 3 years check" },
+        //};
+
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AssessorService.ApplyTypes/Roatp/CriminalCompliancePageConfiguration.cs:14]: Delete rather than comment out.

File: src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs [hunk 7]
```diff
@@ -11,15 +11,67 @@ public static class RoatpGatewayConstants
...
+            //public static string ContractTermination = "ContractTerm";
+            //public static string ContractWithdrawnEarly = "Withdrawn";
+            //public static string Roto = "Roto";
+            //public static string FundingRemoved = "FundingRemoved";
+            //public static string RemovedProfessionalRegister = "RemovedProfessionalRegister";
+            //public static string IttAccreditation = "IttAccreditation";
+            //public static string RemovedCharityRegister = "RemovedCharityRegister";
+            //public static string Safeguarding = "Safeguarding";
+            //public static string Whistleblowing = "Whistleblowing";
+            //public static string Insolvency = "Insolvency";
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs:45]: Delete rather than comment out

File: src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs [hunk 8]
```diff
@@ -11,15 +11,67 @@ public static class RoatpGatewayConstants
...
+        //public static string Roto = "Select the outcome for Removed from Register of Training Organisations(RoTO) in the last 3 years check";
+        //public static string FundingRemoved = "Select the outcome for funding removed from any education bodies in the last 3 years check";
+        //public static string RemovedProfessionalRegister = "Select the outcome for removed from any professional or trade registers in the last 3 years check";
+        //public static string IttAccreditation = "Select the outcome for Involuntary withdrawal from Initial Teacher Training accreditation in the last 3 years check";
+        //public static string RemovedCharityRegister = "Select the outcome for removed from any charity register check";
+        //public static string Safeguarding = "Select the outcome for investigated due to safeguarding issues in the last 3 months check";
+        //public static string Whistleblowing = "Select the outcome for investigated due to whistleblowing issues in the last 3 months check";
+        //public static string Insolvency = "Select the outcome for subject to insolvency or winding up proceedings in the last 3 years check";
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs:75]: Delete rather than comment out.

File: src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/IcoNumberTests.cs [hunk 9]
```diff
@@ -73,7 +73,7 @@ public void check_orchestrator_builds_with_ico_number()
 
             var viewModel = response.Result;
 
-            Assert.AreEqual(UKRLPLegalName, viewModel.UkrlpLegalName);
+            Assert.AreEqual(UKRLPLegalName, viewModel.ApplyLegalName);
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/IcoNumberTests.cs:76]: Not sure this is correct and may need to revert this change. You're now checking the legal name within Apply is equal to that from UKRLP? Is it not possible that it could have changed within UKRLP over time?

File: src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/OrganisationRiskTests.cs [hunk 10]
```diff
@@ -62,7 +62,7 @@ public void Check_organisation_risk_details_are_returned(string organisationType
 
             var viewModel = response.Result;
 
-            Assert.AreEqual(UKRLPLegalName, viewModel.UkrlpLegalName);
+            Assert.AreEqual(UKRLPLegalName, viewModel.ApplyLegalName);
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/OrganisationRiskTests.cs:65]: As mentioned elsewhere regarding name check

File: src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/WebsiteAddressTests.cs [hunk 11]
```diff
@@ -68,7 +68,7 @@ public void check_orchestrator_builds_with_website_address(string organisationWe
 
             var viewModel = response.Result;
 
-            Assert.AreEqual(UKRLPLegalName, viewModel.UkrlpLegalName);
+            Assert.AreEqual(UKRLPLegalName, viewModel.ApplyLegalName);
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/WebsiteAddressTests.cs:71]: As mentioned elsewhere regarding name check

File: src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/RoatpGatewayPageViewModel.cs [hunk 12]
```diff
@@ -13,27 +13,19 @@ public class RoatpGatewayPageViewModel
         public string Status { get; set; }
 
         public string Ukprn { get; set; }
-        public string UkrlpLegalName { get; set; }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/RoatpGatewayPageViewModel.cs:16]: You've removed UkrlpLegalName but I haven't seen this placed in whatever viewmodels & pages need to check UKRLP stuff

File: src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/RoatpGatewayPageViewModel.cs [hunk 13]
```diff
@@ -13,27 +13,19 @@ public class RoatpGatewayPageViewModel
...
         public List<ValidationErrorDetail> ErrorMessages { get; set; }
 
-        public string ApplyLegalName { get; set; }
+   
         public DateTime? SourcesCheckedOn { get; set; }
         public DateTime? ApplicationSubmittedOn { get; set; }
 
         public string Heading { get; set; }
         public string Caption { get; set; }
+        public string NoSelectionErrorMessage { get; set; }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/RoatpGatewayPageViewModel.cs:29]: Should be marked as virtual so that the inherited class can override it.

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationCriminalComplianceChecks.cshtml [hunk 14]
```diff
@@ -119,178 +51,28 @@
...
+                    @Html.HiddenFor(x => x.Caption)
+                    @Html.HiddenFor(x => x.Heading)
+                    @Html.HiddenFor(x => x.NoSelectionErrorMessage)
+                    @Html.HiddenFor(x => x.Ukprn)
+                    @Html.HiddenFor(x => x.ApplyLegalName)
+                    @Html.HiddenFor(x => x.QuestionText)
+                    @Html.HiddenFor(x => x.ComplianceCheckAnswer)
+                    @Html.HiddenFor(x => x.ComplianceCheckQuestionId)
+                    @Html.HiddenFor(x => x.FurtherInformationAnswer)
+                    @Html.HiddenFor(x => x.FurtherInformationQuestionId)
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationCriminalComplianceChecks.cshtml:71]: Use the hidden tag helper as @Html.HiddenFor is deprecated in ASP.NET Core

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationCriminalComplianceChecks.cshtml [hunk 15]
```diff
@@ -119,178 +51,28 @@
...
+                    @{ await Html.RenderPartialAsync("~/Views/Roatp/Apply/Gateway/Pages/_GatewayApprovalSection.cshtml"); }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationCriminalComplianceChecks.cshtml:72]: Use the partial tag helper as Html.RenderPartialAsync is deprecated in ASP.NET Core

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationCriminalComplianceChecks.cshtml [hunk 16]
```diff
@@ -3,104 +3,36 @@
...
-                </div>
-            }
-
-            <span class="govuk-caption-xl">
-                @heading
-            </span>
-
-            <h1 class="govuk-heading-xl">@Model.PageTitle</h1>
-
+            @{ await Html.RenderPartialAsync("~/Views/Roatp/Apply/Gateway/Pages/_GatewayApprovalValidationSummary.cshtml"); }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationCriminalComplianceChecks.cshtml:14]: Use the partial tag helper as Html.RenderPartialAsync is deprecated in ASP.NET Core

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationCriminalComplianceChecks.cshtml [hunk 17]
```diff
@@ -3,104 +3,36 @@
...
-            <a class="govuk-breadcrumbs__link" href="/Roatp/Gateway/New">RoATP applications</a>
-        </li>
-        <li class="govuk-breadcrumbs__list-item">
-            <a class="govuk-breadcrumbs__link" href="/Roatp/Gateway/@Model.ApplicationId">Application assessment overview</a>
-        </li>
-        
-        <li class="govuk-breadcrumbs__list-item" aria-current="page">@Model.PageTitle</li>
-    </ol>
-</div>
+@{ await Html.RenderPartialAsync("~/Views/Roatp/Apply/Gateway/Pages/_GatewayAssessmentBreadcrumb.cshtml"); }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationCriminalComplianceChecks.cshtml:10]: Use the partial tag helper as Html.RenderPartialAsync is deprecated in ASP.NET Core

File: src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs [hunk 18]
```diff
@@ -11,15 +11,67 @@ public static class RoatpGatewayConstants
...
             public static string OrganisationStatusCheck = "Organisation status check";
             public static string AddressCheck = "Address check";
             public static string IcoNumber = "Information Commissioner's Office (ICO) registration number check";
             public static string Website = "Website address check";
             public static string OrganisationRisk = "Organisation high risk check";
+
+            public static string OfficeForStudents = "";
+            public static string InitialTeacherTraining = "";
+            public static string Ofsted = "";
+            public static string SubcontractorDeclaration = "Subcontractor declaration check";
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs:32]: These look like constants to me and should be marked as: public const string

If you need to initialize values at construction, then please mark them as: public static readonly string

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/AddressCheck.cshtml [hunk 19]
```diff
@@ -6,12 +6,12 @@
     Layout = "_Layout";
 }
 
-@{await Html.RenderPartialAsync("~/Views/Roatp/Apply/Gateway/Pages/_GatewayAssessmentBreadcrumb.cshtml"); }
+@{ await Html.RenderPartialAsync("~/Views/Roatp/Apply/Gateway/Pages/_GatewayAssessmentBreadcrumb.cshtml"); }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/AddressCheck.cshtml:9]: Use of deprecated HTML Helper functions. Use tag helpers

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/LegalName.cshtml [hunk 20]
```diff
@@ -3,105 +3,30 @@
...
-            <a class="govuk-breadcrumbs__link" href="/Roatp/Gateway/New">RoATP applications</a>
-        </li>
-        <li class="govuk-breadcrumbs__list-item">
-            <a class="govuk-breadcrumbs__link" href="/Roatp/Gateway/@Model.ApplicationId">Application assessment overview</a>
-        </li>
-        
-        <li class="govuk-breadcrumbs__list-item" aria-current="page">@caption</li>
-    </ol>
-</div>
+@{ await Html.RenderPartialAsync("~/Views/Roatp/Apply/Gateway/Pages/_GatewayAssessmentBreadcrumb.cshtml"); }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/LegalName.cshtml:10]: Use of deprecated HTML Helper functions. Use tag helpers

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationRisk.cshtml [hunk 21]
```diff
@@ -6,22 +6,20 @@
     Layout = "_Layout";
 }
 
-@{await Html.RenderPartialAsync("~/Views/Roatp/Apply/Gateway/Pages/_GatewayAssessmentBreadcrumb.cshtml"); }
+@{ await Html.RenderPartialAsync("~/Views/Roatp/Apply/Gateway/Pages/_GatewayAssessmentBreadcrumb.cshtml"); }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationRisk.cshtml:9]: Use of deprecated HTML Helper functions. Use tag helpers

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/SubcontractorDeclaration.cshtml [hunk 22]
```diff
@@ -87,25 +55,25 @@
...
                 <form class="js-disable-on-submit" asp-action="EvaluateSubcontractorDeclarationPage" asp-controller="RoatpGatewayExperienceAndAccreditation">
-                    <input type="hidden" name="ApplicationId" value="@Model.ApplicationId" />
-                    <input type="hidden" name="PageId" value="@Model.PageId" />
-                    <input type="hidden" name="ApplicationSubmittedOn" value="@Model.ApplicationSubmittedOn" />
-                    <input type="hidden" name="Ukprn" value="@Model.Ukprn" />
-                    <input type="hidden" name="SourcesCheckedOn" value="@Model.SourcesCheckedOn" />
-                    <input type="hidden" name="HasDeliveredTrainingAsSubcontractor" value="@Model.HasDeliveredTrainingAsSubcontractor" />
-                    <input type="hidden" name="ContractFileName" value="@Model.ContractFileName" />
-
+                    @Html.HiddenFor(x => x.ApplicationId)
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/SubcontractorDeclaration.cshtml:65]: Use of deprecated HTML Helper functions. Use tag helpers

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/TradingName.cshtml [hunk 23]
```diff
@@ -3,105 +3,31 @@
...
-            <a class="govuk-breadcrumbs__link" href="/Roatp/Gateway/New">RoATP applications</a>
-        </li>
-        <li class="govuk-breadcrumbs__list-item">
-            <a class="govuk-breadcrumbs__link" href="/Roatp/Gateway/@Model.ApplicationId">Application assessment overview</a>
-        </li>
-        
-        <li class="govuk-breadcrumbs__list-item" aria-current="page">@caption</li>
-    </ol>
-</div>
+@{ await Html.RenderPartialAsync("~/Views/Roatp/Apply/Gateway/Pages/_GatewayAssessmentBreadcrumb.cshtml"); }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/TradingName.cshtml:10]: Use of deprecated HTML Helper functions. Use tag helpers

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/Website.cshtml [hunk 24]
```diff
@@ -59,14 +59,17 @@
...
                 <form class="js-disable-on-submit" asp-action="EvaluateWebsitePage" asp-controller="RoatpGatewayOrganisationChecks">
-                    <input type="hidden" name="ApplicationId" value="@Model.ApplicationId" />
-                    <input type="hidden" name="PageId" value="@Model.PageId" />
-                    <input type="hidden" name="Ukprn" value="@Model.Ukprn" />
-                    <input type="hidden" name="UkrlpLegalName" value="@Model.UkrlpLegalName" />
-                    <input type="hidden" name="ApplicationSubmittedOn" value="@Model.ApplicationSubmittedOn" />
-                    <input type="hidden" name="SourcesCheckedOn" value="@Model.SourcesCheckedOn" />
-                    <input type="hidden" name="SubmittedWebsite" value="@Model.SubmittedWebsite" />
-                    <input type="hidden" name="UkrlpWebsite" value="@Model.UkrlpWebsite" />
+                    @Html.HiddenFor(x => x.ApplicationId)
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/Website.cshtml:62]: Use of deprecated HTML Helper functions. Use tag helpers

File: src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpOrganisationCriminalComplianceChecksController.cs (hunk 2 above)
Comment (David-B-Read) [reply src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpOrganisationCriminalComplianceChecksController.cs:37]: I don't think that should have been commented out

Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpOrganisationCriminalComplianceChecksController.cs:37]: We don't need PageTitle. It's constructed as everywhere else using the heading => ViewBag.Title = $"Gateway - {Model.Heading}";

Comment (David-B-Read) [reply src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpOrganisationCriminalComplianceChecksController.cs:37]: Ok, for criminal compliance pages there are nearly 20 different versions with different titles based on the question ID, so the Model.Heading will need to be populated from that.

Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpOrganisationCriminalComplianceChecksController.cs:37]: Some headings were not populated as Titles as well. That is why I asked for 
“You will also need to populate the rest of the headings in the following file (// TODO Not Assigned)”
 \das-admin-service\src\SFA.DAS.AssessorService.ApplyTypes\Roatp\CriminalCompliancePageConfiguration.cs

File: src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/IcoNumberTests.cs (hunk 9 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/IcoNumberTests.cs:76]: We need to use only ApplyLegalName as common property in RoatpGatewayPageViewModel. And it is the only one populated by PopulatePageCommonDetails.
If someone needs UkrlpLegalName in their particular ViewModel, they can deal with it in their implementation.

File: src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/OrganisationRiskTests.cs (hunk 10 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/OrganisationRiskTests.cs:65]: We need to use only ApplyLegalName as common property in RoatpGatewayPageViewModel. And it is the only one populated by PopulatePageCommonDetails.
If someone needs UkrlpLegalName in their particular ViewModel, they can deal with it in their implementation.

File: src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/WebsiteAddressTests.cs (hunk 11 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/WebsiteAddressTests.cs:71]: We need to use only ApplyLegalName as common property in RoatpGatewayPageViewModel. And it is the only one populated by PopulatePageCommonDetails.
If someone needs UkrlpLegalName in their particular ViewModel, they can deal with it in their implementation.

File: src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/RoatpGatewayPageViewModel.cs (hunk 12 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/RoatpGatewayPageViewModel.cs:16]: We need to use only ApplyLegalName as common property in RoatpGatewayPageViewModel. And it is the only one populated by PopulatePageCommonDetails.
If someone needs UkrlpLegalName in their particular ViewModel, they can deal with it in their implementation.

File: src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/RoatpGatewayPageViewModel.cs (hunk 13 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/RoatpGatewayPageViewModel.cs:29]: Why we need that? I haven't come across such cases, when it needs to be overridden.  I would suggest to be changed only when there is a need for it.

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/RoatpOrganisationCriminalComplianceChecksControllerTests.cs (hunk 1 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/RoatpOrganisationCriminalComplianceChecksControllerTests.cs]: done

File: src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpOrganisationCriminalComplianceChecksController.cs (hunk 2 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpOrganisationCriminalComplianceChecksController.cs]: comment deleted

File: src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayCriminalComplianceChecksOrchestrator.cs (hunk 3 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayCriminalComplianceChecksOrchestrator.cs]: done

File: src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayCriminalComplianceChecksOrchestrator.cs (hunk 4 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayCriminalComplianceChecksOrchestrator.cs]: Now we are passing the values as parameters, but you still will need to remember to set them appropriately. :-)

File: src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/OrganisationCriminalCompliancePageViewModel.cs (hunk 5 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Gateway/OrganisationCriminalCompliancePageViewModel.cs]: done

File: src/SFA.DAS.AssessorService.ApplyTypes/Roatp/CriminalCompliancePageConfiguration.cs (hunk 6 above)
Comment (MakCom) [reply src/SFA.DAS.AssessorService.ApplyTypes/Roatp/CriminalCompliancePageConfiguration.cs]: done

File: src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs (hunk 7 above)
Comment (MakCom) [reply src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs]: done

File: src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs (hunk 8 above)
Comment (MakCom) [reply src/SFA.DAS.AssessorService.ApplyTypes/Roatp/RoatpGatewayConstants.cs]: done

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/LegalName.cshtml (hunk 20 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/LegalName.cshtml:10]: done

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationRisk.cshtml (hunk 21 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/OrganisationRisk.cshtml:9]: done

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/SubcontractorDeclaration.cshtml (hunk 22 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/SubcontractorDeclaration.cshtml]: done

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/TradingName.cshtml (hunk 23 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/TradingName.cshtml:10]: done

File: src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/Website.cshtml (hunk 24 above)
Comment (MakCom) [reply src/SFA.DAS.AdminService.Web/Views/Roatp/Apply/Gateway/Pages/Website.cshtml]: done
